src/reference/antora/modules/ROOT/pages/mail.adoc

```diff
@@ -309,13 +609,112 @@ The following example uses an `imap` URI:
       <int:poller max-messages-per-poll="1" fixed-rate="5000"/>
 </int-mail:inbound-channel-adapter>
 ----
+======
 
 If you do have IMAP `idle` support, you may want to configure the `imap-idle-channel-adapter` element instead.
 Since the `idle` command enables event-driven notifications, no poller is necessary for this adapter.
 It sends a message to the specified channel as soon as it receives the notification that new mail is available.
 The following example configures an IMAP `idle` mail channel:
 
-[source,xml]
+[tabs]
+======
+Java::
++
+[source,java,role="primary"]
+----
+@Bean
+public ImapMailReceiver imapMailReceiver(Properties javaMailProps) {
+    ImapMailReceiver receiver = new ImapMailReceiver("imaps://[username]:[password]@imap.gmail.com/INBOX");
+    receiver.setShouldDeleteMessages(false);
+    receiver.setShouldMarkMessagesAsRead(true);
+    receiver.setJavaMailProperties(javaMailProps);
+    return receiver;
+}
+@Bean
+public ImapIdleChannelAdapter imapIdleChannelAdapter(ImapMailReceiver imapMailReceiver, MessageChannel receiveChannel) {
+    ImapIdleChannelAdapter adapter = new ImapIdleChannelAdapter(imapMailReceiver);
+    adapter.setOutputChannel(receiveChannel);
+    adapter.setAutoStartup(true);
+    adapter.setPhase(Integer.MAX_VALUE);
+    return adapter;
+}
+----
+Java DSL::
++
+[source,java,role="secondary"]
+----
+@Bean
+public IntegrationFlow imapIdleFlow(MessageChannel receiveChannel, MailMessageHandler mailMessageHandler,
+ Properties javaMailProperties) {
+
+	ImapMailReceiver receiver = new ImapMailReceiver("imaps://[username]:[password]@imap.gmail.com/INBOX");
+    receiver.setShouldDeleteMessages(false);
+    receiver.setShouldMarkMessagesAsRead(true);
+    receiver.setJavaMailProperties(javaMailProperties);
+
+    return IntegrationFlow
+        .from(Mail.imapIdleAdapter(receiver)
```

Comment:
Cannot we use `imapIdleAdapter(String url)` and builder of the result of that factory instead of manual `new ImapMailReceiver()` ?